src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -492,6 +492,21 @@ private static void ParseArgumentsIntoList(string arguments, List<string> result
                 // it contains spaces.
                 if (c == '"')
                 {
+                    // While not in quotes, and currentArgument is empty,
```

Comment:
Is the goal here to create a solution for 2.0 or beyond?  If the former, we should make a minimal fix, not replace the implementation wholesale.  If the latter, I agree we should consolidate implementations to whatever extent possible.